x-pack/plugins/alerting/common/alert_schema/field_maps/component_template_from_field_map.ts

```diff
@@ -11,13 +11,15 @@ import { mappingFromFieldMap } from './mapping_from_field_map';
 
 export interface GetComponentTemplateFromFieldMapOpts {
   name: string;
-  fieldLimit?: number;
   fieldMap: FieldMap;
+  includeSettings?: boolean;
+  dynamic?: 'strict' | boolean;
```

Comment:
Added these options in order to more closely match the component templates currently used by the rule registry.

Reply:
Currently we default to `strict` but have component templates specifying `false`. Do we want to type it to only allow `strict` or `false`?

Reply:
Done in https://github.com/elastic/kibana/pull/151792/commits/d3dcbab63a2ce6a21e86caffc2dfecd5b9c2caf5